compiler/src/dotty/tools/repl/ParseResult.scala

```diff
@@ -116,7 +116,8 @@ object ParseResult {
   }
 
   /** Extract a `ParseResult` from the string `sourceCode` */
-  def apply(sourceCode: String)(implicit state: State): ParseResult =
+  def apply(sourceCodeOrFile: Either[String, SourceFile])(implicit state: State): ParseResult = {
```

Comment:
I find Nicolas' suggestion in https://github.com/lampepfl/dotty/pull/6414#discussion_r280096548 clearer than wrapping things in an Either just to rewrap it later.

Reply:
We have the versions for each individual types below. `Either` is to share logic.

Reply:
I get that, but if we always end up wrapping things in a SourceFile, then it makes sense to do that as soon as possible so that we don't need to think about both cases inside the method with all the logic.

Reply:
Ok, good idea